content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/mtls_test.sh

```diff
@@ -22,77 +22,63 @@ set -u
 snip_deploy_a_mutual_tls_server_5
-_wait_for_deployment mesh-external my-nginx
 
-# Generate ServiceEntry and VirtualService for the external service
-snip_deploy_a_mutual_tls_server_6
-_wait_for_istio serviceentry default nginx
-_wait_for_istio virtualservice default nginx
-
-# Generate Secret to hold client certificates
-snip_deploy_a_container_to_test_the_nginx_deployment_1
+# Wait for nginx
+_wait_for_deployment mesh-external my-nginx
 
-# Deploy sleep sample service with certificates mounted and export SOURCE_POD
+# Deploy Sleep example
 set +e
 kubectl delete pods -l app=sleep --force
 set -e
-snip_deploy_a_container_to_test_the_nginx_deployment_2
-_wait_for_deployment default sleep
-snip_deploy_a_container_to_test_the_nginx_deployment_3
-
-# Verify that mTLS connection is set up properly
-_verify_elided snip_deploy_a_container_to_test_the_nginx_deployment_4 "$snip_deploy_a_container_to_test_the_nginx_deployment_4_out"
+snip_deploy_a_mutual_tls_server_6
 
-# Verify that without client certificate request is rejected
-_verify_contains snip_deploy_a_container_to_test_the_nginx_deployment_5 "400 No required SSL certificate was sent"
+_wait_for_deployment default sleep
 
-# Store Client and CA certificates using k8s secret
+# Mount certs to Egress Gateway
 snip_redeploy_the_egress_gateway_with_the_client_certificates_1
-
-# Redeploy and patch Egress Gateway using client cert secrets
 snip_redeploy_the_egress_gateway_with_the_client_certificates_2
 snip_redeploy_the_egress_gateway_with_the_client_certificates_3
+_wait_for_deployment istio-system istio-egressgateway
 
-# TODO: verify tls certs are successfully loaded in istio-egressgateway pod
+# Verify the certs exist
+snip_redeploy_the_egress_gateway_with_the_client_certificates_4
 
-# Direct traffic through egress gateway by creating necessary Gateway, DR, and Virtual Service
+# Open Gateway Listener
 snip_configure_mutual_tls_origination_for_egress_traffic_1
-snip_configure_mutual_tls_origination_for_egress_traffic_2
 _wait_for_istio gateway default istio-egressgateway
 _wait_for_istio destinationrule default egressgateway-for-nginx
+
+# Configure routing from sleep to egress gateway to nginx
+snip_configure_mutual_tls_origination_for_egress_traffic_2
 _wait_for_istio virtualservice default direct-nginx-through-egress-gateway
-_wait_for_istio destinationrule default originate-mtls-for-nginx
 
-# TODO: Verify HTTP connection to nginx
-_verify_elided snip_configure_mutual_tls_origination_for_egress_traffic_3 "$snip_configure_mutual_tls_origination_for_egress_traffic_3_out"
+# Originate TLS with destination rule
+snip_configure_mutual_tls_origination_for_egress_traffic_3
+
+_wait_for_istio destinationrule istio-system originate-mtls-for-nginx
+
+# Verify that mTLS connection is set up properly
+_verify_contains snip_configure_mutual_tls_origination_for_egress_traffic_4 "Welcome to nginx!"
 
 #TODO: verify request is actually being routed through egress gateway
```

Comment:
Why is this a TODO? Doesn't a check similar to the one in `tls_tests.sh` work here?

Reply:
Added this check